hms-app/src/pages/Home.tsx

```diff
@@ -1,6 +1,66 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
+import { Hackathon, HackathonPreview } from '../common/types'
+import {
+  getHackathonDetails,
+  getListOfHackathons,
+} from '../actions/HackathonActions'
+import HackathonSelectDropdown from '../components/HackathonSelectDropdown'
 
 function Home() {
+  const [hackathonList, setHackathonList] = useState<HackathonPreview[]>([])
+  const [isLoading, setIsLoading] = useState(true)
+  const [selectedHackathonId, setSelectedHackathonId] = useState('')
+  const today = new Date()
+  const [hackathonData, setHackathonData] = useState<Hackathon>({
+    id: 'string',
+    title: 'string',
+    startDate: new Date(),
+    endDate: new Date(),
+    participants: [],
+    categories: undefined,
+    ideas: [],
+  })
+
+  const loadHackathons = () => {
+    getListOfHackathons().then((data) => {
+      setHackathonList(data.hackathons)
+      setIsLoading(false)
+    })
+  }
+
+  const loadSelectedHackathon = () => {
+    getHackathonDetails(selectedHackathonId).then((data) => {
+      setHackathonData(data)
+    })
+  }
+
+  const getNextHackathon = hackathonList.filter((hackathon) => {
+    return new Date(hackathon.startDate) > today
+  })
+
+  const timeTillNextHackathon = !isLoading
```

Comment:
when it isLoading it should probably return nothing :thinking:

Reply:
it has to return a number even when its loading